File: app/src/main/java/com/noblemajesty/marvel/models/getCharacters/Result.java

package com.noblemajesty.marvel.models.getCharacters;

import java.util.List;
import android.os.Parcel;
import android.os.Parcelable;
import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class Result implements Parcelable
{

  @SerializedName("id")
  @Expose
  private Integer id;
  @SerializedName("name")
  @Expose
  private String name;
  @SerializedName("description")
  @Expose
  private String description;
  @SerializedName("thumbnail")
  @Expose
  private Thumbnail thumbnail;
  @SerializedName("resourceURI")
  @Expose
  private String resourceURI;
  @SerializedName("comics")
  @Expose
  private Comics comics;
  @SerializedName("series")
  @Expose
  private Series series;
  @SerializedName("stories")
  @Expose
  private Stories stories;
  @SerializedName("events")
  @Expose
  private Events events;
  @SerializedName("urls")
  @Expose
  private List<Url> urls = null;
  public final static Parcelable.Creator<Result> CREATOR = new Creator<Result>() {


    @SuppressWarnings({
        "unchecked"
    })
    public Result createFromParcel(Parcel in) {
      return new Result(in);
    }

    public Result[] newArray(int size) {
      return (new Result[size]);
    }

  }
      ;

  protected Result(Parcel in) {
    this.id = ((Integer) in.readValue((Integer.class.getClassLoader())));
    this.name = ((String) in.readValue((String.class.getClassLoader())));
    this.description = ((String) in.readValue((String.class.getClassLoader())));
    this.thumbnail = ((Thumbnail) in.readValue((Thumbnail.class.getClassLoader())));
    this.resourceURI = ((String) in.readValue((String.class.getClassLoader())));
    this.comics = ((Comics) in.readValue((Comics.class.getClassLoader())));
    this.series = ((Series) in.readValue((Series.class.getClassLoader())));
    this.stories = ((Stories) in.readValue((Stories.class.getClassLoader())));
    this.events = ((Events) in.readValue((Events.class.getClassLoader())));
    in.readList(this.urls, (com.noblemajesty.marvel.models.getCharacters.Url.class.getClassLoader()));
  }

  public Result() {
  }

  public Integer getId() {
    return id;
  }

  public void setId(Integer id) {
    this.id = id;
  }

  public String getName() {
    return name;
  }

  public void setName(String name) {
    this.name = name;
  }

  public String getDescription() {
    return description;
  }

  public void setDescription(String description) {
    this.description = description;
  }


  public Thumbnail getThumbnail() {
    return thumbnail;
  }

  public String getResourceURI() {
    return resourceURI;
  }

  public Comics getComics() {
    return comics;
  }

  public Series getSeries() {
    return series;
  }

  public Stories getStories() {
    return stories;
  }

  public Events getEvents() {
    return events;
  }

  public List<Url> getUrls() {
    return urls;
  }

  public void writeToParcel(Parcel dest, int flags) {
    dest.writeValue(id);
    dest.writeValue(name);
    dest.writeValue(description);
    dest.writeValue(thumbnail);
    dest.writeValue(resourceURI);
    dest.writeValue(comics);
    dest.writeValue(series);
    dest.writeValue(stories);
    dest.writeValue(events);
    dest.writeList(urls);
  }

  public int describeContents() {
    return 0;
  }

}

File: app/src/main/java/com/noblemajesty/marvel/models/getCharacters/Result.kt
package com.noblemajesty.marvel.models.getCharacters

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class Result : Parcelable {

    @SerializedName("id")
    @Expose
    var id: Int? = null
    @SerializedName("name")
    @Expose
    var name: String? = null
    @SerializedName("description")
    @Expose
    var description: String? = null
    @SerializedName("thumbnail")
    @Expose
    var thumbnail: Thumbnail? = null
    @SerializedName("resourceURI")
    @Expose
    var resourceURI: String = ""
    @SerializedName("comics")
    @Expose
    var comics: Comics? = null
    @SerializedName("series")
    @Expose
    var series: Series? = null
    @SerializedName("stories")
    @Expose
    var stories: Stories? = null
    @SerializedName("events")
    @Expose
    var events: Events? = null
    @SerializedName("urls")
    @Expose
    val urls: List<Url>? = null

    protected constructor(`in`: Parcel) {
        this.id = `in`.readValue(Int::class.java.classLoader) as Int
        this.name = `in`.readValue(String::class.java.classLoader) as String
        this.description = `in`.readValue(String::class.java.classLoader) as String
        this.thumbnail = `in`.readValue(Thumbnail::class.java.classLoader) as Thumbnail
        this.resourceURI = `in`.readValue(String::class.java.classLoader) as String
        this.comics = `in`.readValue(Comics::class.java.classLoader) as Comics
        this.series = `in`.readValue(Series::class.java.classLoader) as Series
        this.stories = `in`.readValue(Stories::class.java.classLoader) as Stories
        this.events = `in`.readValue(Events::class.java.classLoader) as Events
        `in`.readList(this.urls, com.noblemajesty.marvel.models.getCharacters.Url::class.java.classLoader)
    }


    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeValue(id)
        dest.writeValue(name)
        dest.writeValue(description)
        dest.writeValue(thumbnail)
        dest.writeValue(resourceURI)
        dest.writeValue(comics)
        dest.writeValue(series)
        dest.writeValue(stories)
        dest.writeValue(events)
        dest.writeList(urls)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object {
        val CREATOR: Parcelable.Creator<Result> = object : Parcelable.Creator<Result> {


            override fun createFromParcel(`in`: Parcel): Result {
                return Result(`in`)
            }

            override fun newArray(size: Int): Array<Result?> {
                return arrayOfNulls(size)
            }

        }
    }

}
